pkg/ccl/kvccl/kvtenantccl/upgradeinterlockccl/upgradeinterlockccl_test: TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC failed

pkg/ccl/kvccl/kvtenantccl/upgradeinterlockccl/upgradeinterlockccl_test.TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12282057?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12282057?buildTab=artifacts#/) on master @ [07915532f9a7def000127029c667c3a91ce90e47](https://github.com/cockroachdb/cockroach/commits/07915532f9a7def000127029c667c3a91ce90e47):


```
=== RUN   TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/54b6bdfc3f2e46888b60afbaf94cbeca/logTestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC4117015748
    test_log_scope.go:81: use -show-logs to present logs inline
    local_test_util_test.go:51: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) upgrade interlock test: running variant "lagging_binary_version", configuration: "pause_after_version_bump_RPC"
    local_test_util_test.go:170: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) creating an initial tenant server
    local_test_util_test.go:178: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) verifying the tenant version
    local_test_util_test.go:182: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) verifying basic SQL functionality
    local_test_util_test.go:187: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) verifying the version of the storage cluster
    local_test_util_test.go:194: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) upgrading the storage cluster
    local_test_util_test.go:197: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) checking the tenant after the storage cluster upgrade
    local_test_util_test.go:201: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) start upgrading the tenant
    local_test_util_test.go:256: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) upgrader is ready
    local_test_util_test.go:258: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) starting another tenant server
    local_test_util_test.go:300: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) shutting down the other tenant server
    local_test_util_test.go:309: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) waiting for the instance table to get in the right state
    local_test_util_test.go:320: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) resuming upgrade
    local_test_util_test.go:322: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) waiting for upgrade to complete
    local_test_util_test.go:324: (TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC) upgrade completed
    local_test_util_test.go:333: 
        	Error Trace:	pkg/ccl/kvccl/kvtenantccl/upgradeinterlockccl/upgradeinterlockccl_test/pkg/ccl/kvccl/kvtenantccl/upgradeinterlockccl/local_test_util_test.go:333
        	            				pkg/ccl/kvccl/kvtenantccl/upgradeinterlockccl/upgradeinterlockccl_test/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvtenantccl/upgradeinterlockccl/generated_test.go:151
        	Error:      	Error "migration-job-find-already-completed: aborted during DistSender.Send: context canceled" does not contain "preventing SQL server from starting because its binary version is too low for the tenant active version"
        	Test:       	TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/54b6bdfc3f2e46888b60afbaf94cbeca/logTestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC4117015748
--- FAIL: TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC (93.95s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112367 pkg/ccl/kvccl/kvtenantccl/upgradeinterlockccl/upgradeinterlockccl_test: TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC failed [C-test-failure O-robot T-multitenant T-testeng branch-release-23.2 release-blocker]
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeInterlock_lagging_binary_version_pause_after_version_bump_RPC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32610

--- a/pkg/ccl/kvccl/kvtenantccl/upgradeinterlockccl/local_test_util_test.go
+++ b/pkg/ccl/kvccl/kvtenantccl/upgradeinterlockccl/local_test_util_test.go
@@ -30,6 +30,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance"
 	"github.com/cockroachdb/cockroach/pkg/sql/stats"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/upgrade/upgradebase"
@@ -94,9 +95,7 @@ func runTest(t *testing.T, variant sharedtestutil.TestVariant, test sharedtestut
 		// the system.sql_instances table) for the failed SQL server to
 		// be removed as quickly as possible. This is because a failed
 		// instance in that table can cause RPC attempts to that server
-		// to fail. The changes below in the other server setup will
-		// ensure that the failed SQL server expires quickly. Here we
-		// need to also ensure that it will be reclaimed quickly.
+		// to fail.
 		instancestorage.ReclaimLoopInterval.Override(ctx, &s.SV, 500*time.Millisecond)
 	}
 
@@ -107,9 +106,16 @@ func runTest(t *testing.T, variant sharedtestutil.TestVariant, test sharedtestut
 		// because a failed instance in that table can cause RPC attempts to
 		// that server to fail. To accomplish this, we decrease the default
 		// TTL and heartbeat for the sessions so that they expire and are
-		// cleaned up faster.
-		slinstance.DefaultTTL.Override(ctx, &s.SV, 250*time.Millisecond)
-		slinstance.DefaultHeartBeat.Override(ctx, &s.SV, 50*time.Millisecond)
+		// cleaned up faster. In cases where this test is being stressed, we
+		// extend the TTL by 10x to handle cases where the system is too
+		// overloaded to heartbeat at sub-second intervals.
+		ttlOverride := 250 * time.Millisecond
+		if skip.Stress() {
+			ttlOverride *= 10
+		}
+		heartbeatOverride := ttlOverride / 10
+		slinstance.DefaultTTL.Override(ctx, &s.SV, ttlOverride)
+		slinstance.DefaultHeartBeat.Override(ctx, &s.SV, heartbeatOverride)
 	}
 
 	// Initialize the version to the BinaryMinSupportedVersion so that

--- a/pkg/sql/sqlliveness/slstorage/slstorage.go
+++ b/pkg/sql/sqlliveness/slstorage/slstorage.go
@@ -540,7 +540,7 @@ func (s *Storage) Insert(
 		s.metrics.WriteFailures.Inc(1)
 		return errors.Wrapf(err, "could not insert session %s", sid)
 	}
-	log.Infof(ctx, "inserted sqlliveness session %s", sid)
+	log.Infof(ctx, "inserted sqlliveness session %s with expiry %s", sid, expiration)
 	s.metrics.WriteSuccesses.Inc(1)
 	return nil
 }

